### PR TITLE
Share link #145

### PR DESCRIPTION
**Pull request by bijanmmarkes**
This completes the request to add a share link and email button to the Modeify Trip Planner. We believe this to be all the requested features and additions to the share link request / branch. This feature was requested to allow employees of VTA to share trips without using their actual email addresses or personal information; along with a quick way to copy the trip to share with others, but anyone who uses our trip planner is allowed access to this feature. A map is also provided in the email of the selected trip route. Please review this code and do your own UAT before approving this pull request. @linuxwebexpert @3vivekb

### CODE REVIEW

File: client/plan/update-routes.js [hunk 1]
```diff
@@ -153,7 +153,8 @@ function updateRoutes(plan, opts, callback) {
         event_type: 'query',
         url: location.href,
         //results: JSON.stringify(data),
-        client_ip: "000.111.000.111",//app.get('client_ip'),
```
Comment (linuxwebexpert) [inline client/plan/update-routes.js:156]: This has never worked properly and was improperly merged to master sometime ago

Any chance we can easily fix this with a MAC ID ? @bijanmmarkes

File: snapshots/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+*
+*/
+!.gitignore
```
Comment (linuxwebexpert) [inline snapshots/.gitignore:3]: Why is this needed ? @bijanmmarkes

Comment (marsxul) [reply snapshots/.gitignore:3]: It probably is a relic of Modeify as a functional TDM solution.  if removing it doesn't impact anything lets get rid of it.  @bijanmmarkes @linuxwebexpert

Comment (linuxwebexpert) [reply snapshots/.gitignore:3]: agreed

Comment (bijanmmarkes) [reply snapshots/.gitignore:3]: In the original .gitignore, It would not let me just ignore the files and keep an empty folder. It required me to create a .gitignore within the actual folder snapshots or it would either have to be completely ignored or uploaded with all my local snapshots. @linuxwebexpert

File: client/plan/update-routes.js (hunk 1 above)
Comment (bijanmmarkes) [reply client/plan/update-routes.js:156]: Not sure if it is an easy fix; As it most certainly will have to be done on the server-side of things, and populated back to the client; And as you know, it would be great if we could do this with a NodeJS back-end module which actually would make it super easy; but unfortunately Component JS is restricting us from those capabilities. @linuxwebexpert

**Comment by 3vivekb** [comment]
- [ ] Mode Choice is missing (Taking bus?)
- [ ] VTA Tripplanner, not VTA Trip Planner should be in the subject line

These can be added as separate issues.

**Comment by 3vivekb** [comment]
- [x] Firefox desktop icons for share and print are off.
